hazelcast/src/main/java/com/hazelcast/util/FutureUtil.java

```diff
@@ -246,6 +264,19 @@ public static void waitWithDeadline(Collection<Future> futures, long timeout, Ti
     }
 
     @PrivateApi
+    public static void waitUntilAllRespondedWithDeadline(Collection<Future> futures, long timeout, TimeUnit timeUnit,
+                                                         ExceptionHandler exceptionHandler) {
+        CollectAllExceptionHandler collector = new CollectAllExceptionHandler(futures.size());
+        waitWithDeadline(futures, timeout, timeUnit, collector);
+        final List<Throwable> throwables = collector.getThrowables();
+        synchronized (throwables) {
```

Comment:
@metanet: can you add a short comment over sync block explaining its reason? You can even add it via direct push after merge. It seems people can be confused here.